Projects/Application/Sources/Common/Extensions/UIViewController+Child.swift

```diff
@@ -0,0 +1,20 @@
+//
+// Created by Igor Tarasenko on 2019-02-11.
+// Licensed under the MIT license
+//
+
+import UIKit
+
+extension UIViewController {
+    func addAndMoveChild(_ viewController: UIViewController) {
+        addChild(viewController)
+        view.addSubview(viewController.view)
+        viewController.didMove(toParent: self)
+    }
+
+    func removeChild(_ viewController: UIViewController) {
```

Comment:
This function seems weird to me. I don't see any usage of the instance on which it was called, but only the instance (`viewController`) that was passed.
Perhaps adding some check like `viewController.parent == self` will clarify things?

Reply:
Fixed